### PR TITLE
docs: document describing how to update symbols map files + bugfix in the symbols generator for major versioning

### PR DESCRIPTION
**Pull request by mattkae**

### CODE REVIEW

File: doc/sphinx/how-to/how-to-update-symbols-map.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+### Scenario 2: Removing or changing a symbol
+1. Make a destructive change to the interface (e.g. remove a parameter from
+   a method).
+2. If not already bumped in this release cycle, bump `MIROIL_ABI``
+   in `src/miroil/CMakeLists.txt`. Set `MIROIL_VERSION_MINOR` and `MIROIL_VERSION_PATCH`
+   to `0`
+3. From the root of the project, run:
+   ```sh
+   cmake --build build --target generate-miral-symbols-map
```
Comment (AlanGriffiths) [inline doc/sphinx/how-to/how-to-update-symbols-map.md:154]: miroil?

**Review by AlanGriffiths — DISMISSED** [review]
Looking good

File: doc/sphinx/how-to/how-to-update-symbols-map.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+- mirwayland
+
+It is tedious to edit these files. Luckily for us, this versioning business
+is (mostly) automated so that we have to think very little about it.
+
+## When are symbols.map files updated
+Two different circumstances will prompt a developer to update
+these files:
+
+1. **A new symbols is added to a library**
```
Comment (tarek-y-ismail) [inline doc/sphinx/how-to/how-to-update-symbols-map.md:27]:
```suggestion
1. **A new symbol is added to a library**
```
or
```suggestion
1. **New symbols are added to a library**
```

File: doc/sphinx/how-to/how-to-update-symbols-map.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+2. **A symbol is changed or removed** (e.g. a new parameter is
+   added to a method call, a class is removed, etc.)
+
+    This is an API break. This means that a consumer of the library
+    can no longer safely compile against the new version of that library
+    without maybe breaking their build. In this scenario, we are forced
+    to create a single new stanza at a brand new version in the `symbols.map`
+    file. This new stanza will contain all of the symbols in that library.
+
+Sometimes we break a symbols but don't even realize it until someone
```
Comment (tarek-y-ismail) [inline doc/sphinx/how-to/how-to-update-symbols-map.md:46]:
```suggestion
Sometimes we change a symbol but don't even realize it until someone
```

File: doc/sphinx/how-to/how-to-update-symbols-map.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+points it out in a pull request. Nowadays, the symbols check is automated
+by CI. You can find results of this check
+[here](https://github.com/canonical/mir/actions/workflows/symbols-check.yml).
+This check is run on each pull request. It will inform you whether or not
+a `symbols.map` file needs to be updated due to your change.
+
+**Warning**: This check is not sophisticated to know if you change the
```
Comment (tarek-y-ismail) [inline doc/sphinx/how-to/how-to-update-symbols-map.md:53]: I can't quite parse what you're trying to say here

Comment (mattkae) [reply doc/sphinx/how-to/how-to-update-symbols-map.md]: Updated! let me know if it reads better to you now

Comment (tarek-y-ismail) [reply doc/sphinx/how-to/how-to-update-symbols-map.md]: Crystal clear now :+1:

File: doc/sphinx/how-to/how-to-update-symbols-map.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+   ```sh
+   cmake --build build --target generate-miral-symbols-map
+   ```
+4. Check that your new symbols is in the `symbols.map` file:
+   ```sh
+   git diff src/miral/symbols.map
+   ```
+5. Regenerate the debian symbols:
+   ```sh
+   cmake --build build --target regenerate-miral-debian-symbols
```
[the remark this answers is not in the document]
Comment (mattkae) [reply doc/sphinx/how-to/how-to-update-symbols-map.md]: Done!

**Review by tarek-y-ismail — APPROVED** [review]

**Pull request merged**